proto/spire/api/registration/registration.proto

```diff
@@ -99,6 +99,52 @@ message EvictAgentResponse {
     spire.common.AttestedNode node = 1;
 }
 
+message MintX509SVIDRequest {
+    // SPIFFE ID of the X509-SVID
+    string spiffe_id = 1;
+
+    // ASN.1 encoded CSR. The CSR is only used to convey the public key and 
+    // prove possession of the private key. The rest of the CSR is ignored.
+    bytes csr = 2;
+
+    // TTL of the X509-SVID. The server default will be used if unset. The TTL
+    // is advisory only. The actual lifetime of the X509-SVID may be lower 
+    // depending on the remaining lifetime of the active SPIRE Server CA.
+    int32 ttl = 3;
```

Comment:
Passing thought on this point: CLI users won't see this comment. Perhaps the CLI tool should print a note at the top of its output when we detect that the TTL we got back is less than what we asked for? I wouldn't want a user to request one year and paste the SVID into a load balancer or something only to have it expire in a week or two

Reply:
Sure, i can do that. It will require parsing the cert and token, but should be ok.